### ISSUE TITLE
exception when reset a repository

### ISSUE THREAD

**Issue by mauricioaniche** [mauricioaniche/metricminer2#15]
stacktrace example

12:47:34 ERROR error processing #0e4e7a62d02e0b6016cdaf019eac874645184425 in /root/android/wordswithcrosses, processor=ck, error=org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of reset command. {0}
java.lang.RuntimeException: org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of reset command. {0}
	at br.com.metricminer2.scm.GitRepository.checkout(GitRepository.java:294)
	at br.com.aniche.fse2016.CKMetrics.process(CKMetrics.java:26)
	at br.com.metricminer2.RepositoryMining.processEverythingOnChangeSet(RepositoryMining.java:164)
	at br.com.metricminer2.RepositoryMining.lambda$mine$0(RepositoryMining.java:100)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of reset command. {0}
	at org.eclipse.jgit.api.ResetCommand.call(ResetCommand.java:227)
	at br.com.metricminer2.scm.GitRepository.checkout(GitRepository.java:288)
	... 8 more
Caused by: org.eclipse.jgit.errors.SymlinksNotSupportedException: Symlinks are not supported with this OS/JRE
	at org.eclipse.jgit.util.FS.createSymLink(FS.java:612)
	at org.eclipse.jgit.dircache.DirCacheCheckout.checkoutEntry(DirCacheCheckout.java:1203)
	at org.eclipse.jgit.dircache.DirCacheCheckout.doCheckout(DirCacheCheckout.java:459)
	at org.eclipse.jgit.dircache.DirCacheCheckout.checkout(DirCacheCheckout.java:394)
	at org.eclipse.jgit.api.ResetCommand.checkoutIndex(ResetCommand.java:393)
	at org.eclipse.jgit.api.ResetCommand.call(ResetCommand.java:200)
	... 9 more

**Comment by mauricioaniche** [comment]
example of repo: https://github.com/scudderfish/MSLoggerBase.git

**Comment by davisjam** [comment]
Can you reproduce this issue? I'm guessing the SCMRepositoryClone code introduced in #98 should fix whatever the bug was.

**Comment by mauricioaniche** [comment]
On my TODO list. However, I am sure this used to happen quite often. For some reason, many repos somehow use _symlinks_, and this crashes the execution. Not sure if there's anything we can do, besides reporting it in our README. I'll be back here.

**Comment by davisjam** [comment]
Were you doing this on Windows? [Relevent](https://stackoverflow.com/questions/11662868/what-happens-when-i-clone-a-repository-with-symlinks-on-windows)?

**Comment by mauricioaniche** [comment]
I'm a Mac user!! :)

**Comment by davisjam** [comment]
Possibly relevant [open bug report](https://ecosystem.atlassian.net/browse/MJF-270) against JGit.

**Comment by klerisson** [comment]
Just face similar situation:

java.lang.RuntimeException: org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of reset command. {0}
	at org.repodriller.scm.GitRepository.checkout(GitRepository.java:369)
	at br.ufu.facom.lascam.metricextractor.visitor.RevisionVisitor.process(RevisionVisitor.java:30)
	at org.repodriller.RepositoryMining.processChangeSet(RepositoryMining.java:180)
	at org.repodriller.RepositoryMining.lambda$processRepos$5(RepositoryMining.java:116)
	at org.repodriller.RepositoryMining$$Lambda$1/1092572064.run(Unknown Source)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1142)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:617)
	at java.lang.Thread.run(Thread.java:745)
Caused by: org.eclipse.jgit.api.errors.JGitInternalException: Exception caught during execution of reset command. {0}
	at org.eclipse.jgit.api.ResetCommand.call(ResetCommand.java:227)
	at org.repodriller.scm.GitRepository.checkout(GitRepository.java:363)
	... 9 more
Caused by: org.eclipse.jgit.errors.LockFailedException: Cannot lock /home/klerisson/Documents/PhD/research_artifacts/experiment/storm/.git/index
	at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:224)
	at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:301)
	at org.eclipse.jgit.dircache.DirCache.lock(DirCache.java:267)
	at org.eclipse.jgit.lib.Repository.lockDirCache(Repository.java:1053)
	at org.eclipse.jgit.api.ResetCommand.checkoutIndex(ResetCommand.java:387)
	at org.eclipse.jgit.api.ResetCommand.call(ResetCommand.java:200)
	... 10 more

**Comment by klerisson** [comment]
Works for me: [Stack Overflow](https://stackoverflow.com/questions/43995106/how-to-solved-this-exception-caught-during-execution-of-merge-command-org-ecli)

**Comment by mauricioaniche** [comment]
Yes, sometimes that happens when Git itself gets lost. The manual solution is indeed to delete the lock manually. However, I don't see how to put it on RepoDriller, as I'm afraid of other consequences (i.e. unexpected behavior, like collecting bad data) it might bring to the tool.
